arrow-udf-python/src/lib.rs

```diff
@@ -198,19 +212,99 @@ impl Runtime {
         handler: &str,
     ) -> Result<()> {
         let function = self.interpreter.with_gil(|py| {
-            Ok(PyModule::from_code_bound(py, code, "", "")?
+            Ok(PyModule::from_code_bound(py, code, name, name)?
                 .getattr(handler)?
                 .into())
         })?;
         let function = Function {
             function,
-            return_field: return_type.into_field(name),
+            return_field: return_type.into_field(name).into(),
             mode,
         };
         self.functions.insert(name.to_string(), function);
         Ok(())
     }
 
+    /// Add a new aggregate function from Python code.
+    ///
+    /// # Arguments
+    ///
+    /// - `name`: The name of the function.
+    /// - `state_type`: The data type of the internal state.
+    /// - `output_type`: The data type of the aggregate value.
+    /// - `mode`: Whether the function will be called when some of its arguments are null.
+    /// - `code`: The Python code of the aggregate function.
+    ///
+    /// The code should define at least two functions:
+    ///
+    /// - `create_state() -> state`: Create a new state object.
+    /// - `accumulate(state, *args) -> state`: Accumulate a new value into the state, returning the updated state.
+    ///
+    /// optionally, the code can define:
+    ///
+    /// - `finish(state) -> value`: Get the result of the aggregate function.
+    ///     If not defined, the state is returned as the result. `output_type` must equal to `state_type`.
+    /// - `retract(state, *args) -> state`: Retract a value from the state, returning the updated state.
+    /// - `merge(state, state) -> state`: Merge two states, returning the merged state.
```

Comment:
Does RisingWave actually use `merge()`? If yes, in which situation?

Reply:
It will be used in the second phase of 2-phase aggregation. But for our existing built-in aggregates, their merge function is either themselves (sum -> sum) or another function (count -> sum). So we don't have the concept of an explicit `merge` function yet. But this is necessary after we introduce UDAF.